code/assurance/src/main/java/com/adobe/marketing/mobile/assurance/internal/AssuranceSession.java

```diff
@@ -500,9 +500,15 @@ private void notifySessionDisconnection(
      */
     private void notifyTerminationAndRemoveStatusListeners(
             @Nullable AssuranceConstants.AssuranceConnectionError error) {
-        for (final AssuranceSessionStatusListener listener : sessionStatusListeners) {
+        // Notify about disconnection to all the listeners.
+        notifySessionDisconnection(error);
+
+        // Unregister all the listeners. Operate on a copy of the listeners as
+        // unregisterStatusListener modifies the listeners set.
+        final Set<AssuranceSessionStatusListener> listenersCopy =
```

Comment:
How about 
sessionStatusListeners.clear()

Reply:
Yes! Clearing here is better. Will patch.